src/string_array.c

```diff
@@ -47,19 +50,13 @@ utilities_string_array_fini(string_array_t * array)
 
   utilities_ret_t ret = UTILITIES_RET_OK;
   for (size_t i = 0; i < array->size; ++i) {
-    if (array->data[i]) {
-      free(array->data[i]);
-      array->data[i] = NULL;
-    } else {
-      ret = UTILITIES_RET_WARN;
-    }
+    free(array->data[i]);
+    array->data[i] = NULL;
   }
 
   if (array->data) {
```

Comment:
This should be checked before you loop over `array->data` in the previous for loop.